### PR TITLE
✨  sast cleanup

### PR DESCRIPTION
**Pull request by laurentsimon**
* **Please check if the PR fulfills these requirements**
- [] Tests for the changes have been added (for bug fixes / features)
- [X] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
cleanup

* **What is the current behavior?** (You can also link to an open issue here)
messages can be improved.

* **What is the new behavior (if this is a feature change)?**
cleanup sast check. This PR just improves messaging returend by check.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: checks/sast.go [hunk 1]
```diff
@@ -61,36 +62,39 @@ func SAST(c *checker.CheckRequest) checker.CheckResult {
...
-			return checker.CreateProportionalScoreResult(CheckSAST,
-				"all commmits are checked with a SAST tool", sastScore, checker.MaxResultScore)
-		case sastScore != checker.MaxResultScore && sastScore >= codeQlScore:
-			return checker.CreateMinScoreResult(CheckSAST,
-				"no SAST tool detected")
+		case sastScore == checker.MaxResultScore:
+			return checker.CreateMaxScoreResult(CheckSAST, "a SAST tool is run on all commits")
+		case codeQlScore == checker.MinResultScore:
+			return checker.CreateResultWithScore(CheckSAST,
+				checker.NormalizeReason("no SAST is run on all commits", sastScore), sastScore)
```
Comment (inferno-chromium) [inline checks/sast.go:69]: slightly confusing to read.
maybe "SAST tool is not run on all commits" ?

File: checks/sast.go [hunk 2]
```diff
@@ -61,36 +62,39 @@ func SAST(c *checker.CheckRequest) checker.CheckResult {
...
-	// Only happens if no merges are detected.
+	// CodeQl inconclusive.
 	if sastScore != checker.InconclusiveResultScore {
-		return checker.CreateResultWithScore(CheckSAST, "no merges detected", sastScore)
+		if sastScore == checker.MaxResultScore {
+			return checker.CreateMaxScoreResult(CheckSAST, "a SAST tool is run on all commits")
+		}
+
+		return checker.CreateResultWithScore(CheckSAST,
+			checker.NormalizeReason("no SAST is run on all commits", sastScore), sastScore)
```
Comment (inferno-chromium) [inline checks/sast.go:97]: slightly confusing to read.
maybe "SAST tool is not run on all commits" ?

**Review by inferno-chromium — APPROVED** [review]

**Review by oliverchang — APPROVED** [review]

File: checks/sast.go (hunk 1 above)
Comment (laurentsimon) [reply checks/sast.go]: done

File: checks/sast.go (hunk 2 above)
Comment (laurentsimon) [reply checks/sast.go]: done.

**Pull request merged**